src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py

```diff
@@ -1140,6 +1140,58 @@ def test_functionapp_with_app_insights_key(self, resource_group, storage_account
         self.cmd('functionapp delete -g {} -n {}'.format(resource_group, functionapp_name))
 
 
+class FunctionAppWithAppInsightsDefault(ScenarioTest):
+    @ResourceGroupPreparer(location='westus')
+    @StorageAccountPreparer()
+    def test_functionapp_with_default_app_insights(self, resource_group, storage_account):
+        functionapp_name = self.create_random_name('functionappwithappinsights', 40)
+
+        self.cmd('functionapp create -g {} -n {} -c westus -s {} --os-type Windows'
+                 .format(resource_group, functionapp_name, storage_account)).assert_with_checks([
+                     JMESPathCheck('state', 'Running'),
+                     JMESPathCheck('name', functionapp_name),
+                     JMESPathCheck('kind', 'functionapp'),
+                     JMESPathCheck('hostNames[0]', functionapp_name + '.azurewebsites.net')])
+
+        app_set = self.cmd('functionapp config appsettings list -g {} -n {}'.format(resource_group, functionapp_name)).get_output_in_json()
+        self.assertTrue('APPINSIGHTS_INSTRUMENTATIONKEY' in [kp['name'] for kp in app_set])
+        self.cmd('functionapp delete -g {} -n {}'.format(resource_group, functionapp_name))
+
+    @ResourceGroupPreparer(location='westus')
+    @StorageAccountPreparer()
+    def test_functionapp_with_no_default_app_insights(self, resource_group, storage_account):
+        functionapp_name = self.create_random_name('functionappwithappinsights', 40)
+
+        self.cmd('functionapp create -g {} -n {} -c centralus -s {} --os-type Windows --app-insights-disable'
+                 .format(resource_group, functionapp_name, storage_account)).assert_with_checks([
+                     JMESPathCheck('state', 'Running'),
+                     JMESPathCheck('name', functionapp_name),
+                     JMESPathCheck('kind', 'functionapp'),
+                     JMESPathCheck('hostNames[0]', functionapp_name + '.azurewebsites.net')])
+
+        app_set = self.cmd('functionapp config appsettings list -g {} -n {}'.format(resource_group,
+                                                                                    functionapp_name)).get_output_in_json()
+        self.assertTrue('APPINSIGHTS_INSTRUMENTATIONKEY' not in [kp['name'] for kp in app_set])
+        self.cmd('functionapp delete -g {} -n {}'.format(resource_group, functionapp_name))
```

Comment:
To get test run fast, I suggest skip this since `ResourceGroupPreparer` would clean them up anyway asynchronously, unless you do want to test the `delete`

Reply:
That's good to know. Thanks Yugang!